build/build_android_llm_demo.sh

```diff
@@ -0,0 +1,140 @@
+  cp extension/android/build/libs/executorch.jar "${BUILD_AAR_DIR}/libs/"
+}
+
+build_android_native_library() {
+  ANDROID_ABI="$1"
+  TOKENIZER="$2"
+  ANDROID_NDK="${ANDROID_NDK:-/opt/ndk}"
+  CMAKE_OUT="cmake-out-android-${ANDROID_ABI}"
+  if [[ $TOKENIZER == "tiktoken" ]]; then
+    EXECUTORCH_USE_TIKTOKEN=ON
+  else
+    EXECUTORCH_USE_TIKTOKEN=OFF
+  fi
+
+  cmake . -DCMAKE_INSTALL_PREFIX="${CMAKE_OUT}" \
+    -DCMAKE_TOOLCHAIN_FILE="${ANDROID_NDK}/build/cmake/android.toolchain.cmake" \
+    -DANDROID_ABI="${ANDROID_ABI}" \
+    -DANDROID_PLATFORM=android-23 \
+    -DEXECUTORCH_BUILD_XNNPACK=ON \
+    -DEXECUTORCH_BUILD_EXTENSION_DATA_LOADER=ON \
+    -DEXECUTORCH_BUILD_EXTENSION_MODULE=ON \
+    -DEXECUTORCH_BUILD_KERNELS_OPTIMIZED=ON \
+    -DEXECUTORCH_BUILD_KERNELS_QUANTIZED=ON \
+    -DEXECUTORCH_BUILD_KERNELS_CUSTOM=ON \
+    -DCMAKE_BUILD_TYPE=Release \
+    -B"${CMAKE_OUT}"
+
+  if [ "$(uname)" == "Darwin" ]; then
+    CMAKE_JOBS=$(( $(sysctl -n hw.ncpu) - 1 ))
+  else
+    CMAKE_JOBS=$(( $(nproc) - 1 ))
+  fi
+  cmake --build "${CMAKE_OUT}" -j "${CMAKE_JOBS}" --target install --config Release
+
+  cmake examples/models/llama2 \
+    -DCMAKE_TOOLCHAIN_FILE=$ANDROID_NDK/build/cmake/android.toolchain.cmake \
+    -DANDROID_ABI="$ANDROID_ABI" \
+    -DANDROID_PLATFORM=android-23 \
+    -DCMAKE_INSTALL_PREFIX="${CMAKE_OUT}" \
+    -DEXECUTORCH_USE_TIKTOKEN="${EXECUTORCH_USE_TIKTOKEN}" \
+    -DEXECUTORCH_BUILD_KERNELS_CUSTOM=ON \
+    -DEXECUTORCH_BUILD_KERNELS_OPTIMIZED=ON \
+    -DEXECUTORCH_BUILD_XNNPACK=ON \
+    -DCMAKE_BUILD_TYPE=Release \
+    -B"${CMAKE_OUT}"/examples/models/llama2
+
+  cmake --build "${CMAKE_OUT}"/examples/models/llama2 -j "${CMAKE_JOBS}" --config Release
+
+  cmake extension/android \
+    -DCMAKE_TOOLCHAIN_FILE=${ANDROID_NDK}/build/cmake/android.toolchain.cmake \
+    -DANDROID_ABI="${ANDROID_ABI}" \
+    -DANDROID_PLATFORM=android-23 \
+    -DCMAKE_INSTALL_PREFIX="${CMAKE_OUT}" \
+    -DEXECUTORCH_BUILD_LLAMA_JNI=ON \
+    -DEXECUTORCH_USE_TIKTOKEN="${EXECUTORCH_USE_TIKTOKEN}" \
+    -DCMAKE_BUILD_TYPE=Release \
+    -B"${CMAKE_OUT}"/extension/android
+
+  cmake --build "${CMAKE_OUT}"/extension/android -j "${CMAKE_JOBS}" --config Release
+
+  # Copy artifacts to ABI specific directory
+  mkdir -p "${BUILD_AAR_DIR}/jni/${ANDROID_ABI}"
+  cp "${CMAKE_OUT}"/extension/android/*.so "${BUILD_AAR_DIR}/jni/${ANDROID_ABI}/"
+}
+
+build_aar() {
+  echo \<manifest xmlns:android=\"http://schemas.android.com/apk/res/android\" \
+  package=\"org.pytorch.executorch\"\> \
+  \<uses-sdk android:minSdkVersion=\"19\" /\> \
+  \</manifest\> > "${BUILD_AAR_DIR}/AndroidManifest.xml"
+  pushd "${BUILD_AAR_DIR}"
+  # Rename libexecutorch_jni.so to libexecutorch.so for soname consistency
+  # between Java and JNI
+  find jni -type f -name "libexecutorch_jni.so" -exec bash -c 'mv "$1" "${1/_jni/}"' bash {} \;
+  # Zip all necessary files into the AAR file
+  zip -r executorch.aar libs jni/*/libexecutorch.so AndroidManifest.xml
```

Comment:
@kirklandsign Is this for llama demo or for the other demo app?

Reply:
This one is for the other demo app (image processing)

Reply:
Ok, I will split it out. Are there other things for the vision model demo?

Reply:
We need to publish two aars (llama and other model) for now, because the API are very unsimilar. However, the underlying ops and backend are the same. So in the future, might worth using 1 instead to have both functionalities, but the size will be larger

Reply:
> Ok, I will split it out. Are there other things for the vision model demo?

Nothing else. Just different .so for these two .aar

Reply:
Size should not be a concern since we are not deploy it for production. Test efficiency is our consideration. So what's your suggestion? Should we keep aars for all demos together? Imagine in the future we have more demo apps with different user interfaces.

Reply:
I see. For now let's just use two separate aars like now, because I don't feel like it blocks our test efficiency, and having two .so files with duplicated backends and kernels is really wasting space. So let's land this PR for now. I will try building a single .so file for llama and non-llama then. I probably need to do some refactor because right now we have two entry points to load the .so files from Java for llama and non-llama.